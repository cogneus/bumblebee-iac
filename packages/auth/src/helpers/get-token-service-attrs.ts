import { env } from 'process'
import { TokenServiceAttrs } from '../types'

export const getTokenServiceAttrs = (): TokenServiceAttrs => {
  const { TOKEN_AUDIENCE, TOKEN_EXPIRY, TOKEN_ISSUER } = env
  return {
    audience: TOKEN_AUDIENCE,
    expiresIn: TOKEN_EXPIRY,
    issuer: TOKEN_ISSUER,
  }
}
