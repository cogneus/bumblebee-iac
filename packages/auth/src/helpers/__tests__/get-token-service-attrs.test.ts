import { getTokenServiceAttrs } from '../get-token-service-attrs'

describe('Get token service attrs tests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
    process.env = {
      TOKEN_AUDIENCE: 'audience',
      TOKEN_EXPIRY: 'expiry',
      TOKEN_ISSUER: 'issuer',
    } as any
  })

  test('should return the value of the secrets key', async () => {
    const value = getTokenServiceAttrs()
    expect(value).toMatchSnapshot()
  })
})
