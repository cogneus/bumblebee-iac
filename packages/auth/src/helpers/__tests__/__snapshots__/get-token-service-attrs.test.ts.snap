// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get token service attrs tests should return the value of the secrets key 1`] = `
{
  "audience": "audience",
  "expiresIn": "expiry",
  "issuer": "issuer",
}
`;
